### DISCUSSION TITLE
Localization

### DISCUSSION THREAD

**Discussion by gizburdt** [livewire/livewire#1143] [4. General]
Hi, first of all. Damn good job on Livewire!

I'm using a package for extra localization based on url (/en/, /de/, /nl/). This works until I click a wired element that refreshed the livewire component. After that, Livewire/Laravel doesn't know which locale is used, since the call to the Livewire controller doesn't include a route parameter that determines the locale. 

The package I'm using is: https://github.com/mcamara/laravel-localization

Do you maybe have an idea how to tackle something like this?

**Comment by ivanvermeyen** [comment]
Hello,

I have the same problem with my own package.
Although the middleware is setting the App locale, Livewire doesn't seem to pick it up.
I wrote down my findings in more detail in the following issue and created a test app (linked in the issue):

https://github.com/codezero-be/laravel-localized-routes/issues/58#issuecomment-1497402190

Do I have to make Livewire aware of the locale in some way?

Edit:

The problem seems to be caused by temporarily updating the App locale when generating the URLs for alternative locales.
I do this to automatically fetch translated route parameters.

https://github.com/codezero-be/laravel-localized-routes/blob/2765567c0b99e4ed079630ad8d188bd65fae72ef/src/Illuminate/Routing/UrlGenerator.php#L30-L40

If I comment out the 2 calls to `App::setLocale()`, the locale is correct in Livewire and the `{locale}/livewire/message/{name}` endpoint is used for the ajax calls.

I didn't find yet why Livewire ignores the second call, when I restore the active locale.
Is there something async happening when the app locale changes?

Edit 2:

Turns out it's a bug in my code.
When generating a route throws an exception, the locale doesn't get restored. 
This is not a problem until I try/catch that process and the wrong locale stays active :)
I can at least fix it now :)
